cpp/src/strings/convert/convert_fixed_point.cu

```diff
@@ -0,0 +1,450 @@
+    auto d_results = results->mutable_view().data<DecimalType>();
+
+    // convert strings into decimal values
+    thrust::transform(rmm::exec_policy(stream),
+                      thrust::make_counting_iterator<size_type>(0),
+                      thrust::make_counting_iterator<size_type>(input.size()),
+                      d_results,
+                      string_to_decimal_fn<DecimalType>{*d_column, output_type.scale()});
+    results->set_null_count(input.null_count());
+    return results;
+  }
+
+  template <typename T, std::enable_if_t<not cudf::is_fixed_point<T>()>* = nullptr>
+  std::unique_ptr<column> operator()(strings_column_view const&,
+                                     data_type,
+                                     rmm::cuda_stream_view,
+                                     rmm::mr::device_memory_resource*) const
+  {
+    CUDF_FAIL("Output for to_fixed_point must be a decimal type.");
+  }
+};
+
+}  // namespace
+
+// This will convert a strings column into any integer column type.
+std::unique_ptr<column> to_fixed_point(strings_column_view const& input,
+                                       data_type output_type,
+                                       rmm::cuda_stream_view stream,
+                                       rmm::mr::device_memory_resource* mr)
+{
+  if (input.is_empty()) return make_empty_column(output_type);
+  return type_dispatcher(output_type, dispatch_to_fixed_point_fn{}, input, output_type, stream, mr);
+}
+
+}  // namespace detail
+
+// external API
+std::unique_ptr<column> to_fixed_point(strings_column_view const& strings,
+                                       data_type output_type,
+                                       rmm::mr::device_memory_resource* mr)
+{
+  CUDF_FUNC_RANGE();
+  return detail::to_fixed_point(strings, output_type, rmm::cuda_stream_default, mr);
+}
+
+namespace detail {
+namespace {
+/**
+ * @brief Calculate the size of the each string required for
+ * converting each value in base-10 format.
+ *
+ * ouput format is [-]integer.fraction
+ */
+template <typename DecimalType>
+struct decimal_to_string_size_fn {
+  column_device_view const d_column;
+
+  __device__ int32_t operator()(size_type idx) const
+  {
+    if (d_column.is_null(idx)) return 0;
+    auto const value = d_column.element<DecimalType>(idx);
+    auto const scale = d_column.type().scale();
+
+    if (scale >= 0) return count_digits(value) + scale;
+
+    auto const abs_value = std::abs(value);
+    auto const exp_ten   = static_cast<int32_t>(exp10(static_cast<double>(-scale)));
+    auto const num_zeros = std::max(0, (-scale - count_digits(abs_value % exp_ten)));
+    return static_cast<int32_t>(value < 0) +    // sign if negative
+           count_digits(abs_value / exp_ten) +  // integer
+           1 +                                  // decimal point
+           num_zeros +                          // zeros padding
+           count_digits(abs_value % exp_ten);   // fraction
```

Comment:
How much do we trust nvcc to optimize-out the seperate calls to `count_digits(abs_value % exp_ten)` ?

Reply:
I don't understand. They are unique calls. One for integer component and one for the remainder.

Reply:
This call:  `count_digits(abs_value % exp_ten)`  happens twice.

Reply:
Ah good catch. I missed that.